Assets/Scripts/Utilities/GrossUtils.cs

```diff
@@ -65,27 +65,15 @@ public void OnAfterDeserialize()
 		// Probably adding an item in the inspector
 		if ( keys.Count + 1 == values.Count )
 		{
-			if ( keys.Count == 0 )
-			{
-				keys.Add( new TKey() );
-			}
-			else
-			{
-				keys.Add( keys[keys.Count - 1] ); // duplicate the last key
-			}
+			keys.Add( new TKey() );
 		}
 
 		// Probably adding an item in the inspector
 		if ( values.Count + 1 == keys.Count )
 		{
-			if ( values.Count == 0 )
-			{
-				values.Add( new TValue() );
-			}
-			else
-			{
-				values.Add( values[values.Count - 1] ); // duplicate the value key
-			}
+			keys[keys.Count - 1] = new TKey();// If the keys list is extended in the inspector, it will copy the last key in the list
```

Comment:
The problem was that Unity automatically duplicates the last item in a list when you increase the size. So if you increased the size of the keys list it copies the last key, and THEN you have duplicate keys. So I'm catching that case here and setting that last item in the list to a new key that is the result of the default constructor.

Should all that be made clearer in the code?